### PR TITLE
EnggGUI: Fixes and improvements for fitting tab

### LINKED ISSUE #16194: EnggGUI: fixes and improvements for fitting plot
After feedback form `ENGIN-X` scientist, the plot on the `Fitting` tab required some changes.

- The plot of the `focused workspace` should be plotted in crosses instead of steady line
- The peaks should be more visible than `focused workspace`
-  `focused workspace` should still be displayed when the fitting fails
- Add clear button to clear the list of peak list
- Improvement to the layout of tab
- button to plot single peak fitting workspace in separate window

**Bug:** *Small fitting tab related issue: when file is selecting using browse button, the correct bank file is not selected*

### PR DESCRIPTION
**Pull request by shahrozahmed1**
**Description of work.**

Various changes which were required on the `Fitting` tab after feedback from `Engin-X` scientists.
The plot of the focused workspace should now be plotting in crosses instead of lines, the focused workspace in the background should now be plotted in dark grey to improve to visibility of the single fit peaks. When fitting fails, the focused workspace should still be plotted in the background in `dSpacing` so that peaks on top can still be added. When file is selecting using browse button, the correct bank file should now be selected.
Improved layout of the tab; the peak tools to plot add new peaks have been differentiated in to separate group. `Save` and `Clear` button have been introduced next to the peak list.
A `Plot In Separate Window` button which will enable user to plot all peaks in `engggui_fitting_single_peaks` workspace

**To test:**

Go to: `Interfaces/Diffraction/Engineering Diffraction/Fitting`
`Browse` to the file and make sure the correct file selected is found within Focused Run# text-field

_Would be good idea testing invalid focused file e.g:_ (`ENGINX_228062_focused_bank_1`) _can be found under:_  `\\olympic\Babylon5\Public\Shahroz\FittingTest` 

Click `Clear` button to check whether the peaks list is cleared
Click `Save` button to check the peaks list is still being saved 
Input invalid peak and click `Fit`
Check that focused workspace is always displayed whether the fittings fails or not
Check that focused workspace is being plotted in `dSpacing` so that valid `dSpacing` peaks can be added to list
Check that `Plot In Separate Window` is disabled when fitting fails
This can be tested by: Selecting peak using clicking `Select Peak` and adding to list by clicking `Add Peak`
Click Fit and Check the peaks added have been fitted
Click `Plot In Separate Window` to check if `engggui_fitting_single_peaks` Workspace is plotted with appropriate title

Fixes #16194

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the coding standards? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [x] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by FedeMPouzols** [comment]
There are a few doxygen issues in the view files.

**Comment by shahrozahmed1** [comment]
@FedeMPouzols PR is now ready and test has been updated & please let me know if `Plot To Separate Window` button should be located elsewhere on the tab

File: MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp [hunk 1]
```diff
@@ -1327,6 +1362,32 @@ std::string EnggDiffractionViewQtGUI::getFittingRunNo() const {
...
+      "\"engggui_fitting_single_peaks\", OutputWorkspace = "
+      "fitting_single_peaks_twin_ws)\n"
+      "tot_spec = single_peak_ws.getNumberHistograms()\n"
+
+      "spec_list = []\n"
+      "for i in range(0, tot_spec):\n"
+      " spec_list.append(i)\n"
+
+      "fitting_plot = plotSpectrum(single_peak_ws, spec_list).activeLayer()\n"
+      "fitting_plot.setTitle(\"Engg GUI Single Peaks Fitting Workspace\")\n";
```
Comment (FedeMPouzols) [inline MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp]: Let's keep an eye on this type of script-building code that should be moved to the presenter.

File: MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp [hunk 2]
```diff
@@ -1581,6 +1642,11 @@ void MantidQt::CustomInterfaces::EnggDiffractionViewQtGUI::savePeakList() {
   }
 }
 
+void MantidQt::CustomInterfaces::EnggDiffractionViewQtGUI::clearPeakList() {
+  QString emptyStr = "";
+  m_uiTabFitting.lineEdit_fitting_peaks->setText(emptyStr);
```
Comment (FedeMPouzols) [inline MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp]: Here you can use `QLineEdit::clear()`.

**Comment by FedeMPouzols** [comment]
This looks good. I'm going to do now the functional tests...

**Comment by FedeMPouzols** [comment]
Remember to state if this requires release notes or not.

**Comment by FedeMPouzols** [comment]
Modifies the style of the lines in the "peaks fitting" plot according to instrument scientists preferences. I think it looks good, and the style is much more convenient than the GSAS plots for example.

This also makes sure that the input spectrum is plotted, even when the peak fitting algorithm fails because of poor data, poor initialization, etc.

This works well and makes the interface more usable:
- When one or more peaks are fitted correctly the plots look good.
- The Clear button behaves as expected
- The Save button saves files correctly, handles empty and non-empty lists. Then browse can load them correctly.
- Data and/or initial peak centers that put EnggFitPeaks in trouble: the fitting algorithm fails but the plot shows the spectrum correctly and the messages are informative. Good. When this fails the "plot to separate window" remains disabled which makes sense.
- Well behaved data and initial centers: plotted correctly. The "plot to separate window" button works as expected.
- The documentation has been extended and there a release note bullet point for this.

This finishes all the issues specific to the fitting tab and the list of peaks identified during the "early beta testing" / demo session of last week. The next batch of fixes should follow hopefully early next week: #16252.

:shipit:

**Comment by martyngigg** [comment]
Pure GUI changes so no need to wait for system tests to complete.

**Pull request merged**